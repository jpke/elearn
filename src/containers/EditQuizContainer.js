import React from 'react';
import {connect} from 'react-redux';
import {createQuizView, createQuiz, addItem} from '../actions/eLearnActions';
import EditQuizItemsContainer from './EditQuizItemsContainer';
import CreateQuizView from '../components/CreateQuizView';

export const EditQuizContainer = (props) => {
  if(props.message) {
    alert(props.message);
  }
  const createQuiz = (event) => {
    event.preventDefault();
    const form = event.target.elements;
    props.createQuiz(props.token, form.title.value, props.courseId, form.minimumScore.value);
  };

  return <EditQuizItemsContainer />
};

// QuizPage.propTypes

function mapStateToProps(state) {
  return {
    newQuiz: state.quizReducer.newQuiz,
    courseId: state.authReducer.course._id,
    token: state.authReducer.token,
    quizSelected: state.quizReducer.quizSelected,
    quizSelectedId: state.quizReducer.quizSelectedId,
    message: state.quizReducer.message || ""
  };
}
function mapDispatchToProps(dispatch) {
  return {
    createQuizViewToggle: () => dispatch(createQuizView()),
    addItem: (item) => dispatch(addItem(item)),
    createQuiz: (token, title, courseId, minScore) => dispatch(createQuiz(token, title, courseId, minScore))
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditQuizContainer);
