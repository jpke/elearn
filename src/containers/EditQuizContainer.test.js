import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
const store = configureStore();

import EditQuizContainer from './EditQuizContainer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <EditQuizContainer />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
