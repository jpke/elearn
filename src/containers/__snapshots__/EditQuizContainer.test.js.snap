exports[`test renders correctly 1`] = `
<div
  className="editQuiz-container">
  <div
    className="backToQuizList">
    <div />
    <button
      className="toggleQuizView"
      id="toggleQuizView"
      onClick={[Function]}>
      Back to Quizzes
    </button>
    <div>
      <h3 />
    </div>
  </div>
  <div
    className="editQuizTitle">
    <h3>
      Title
    </h3>
    <input
      className="edit-quiz-item"
      id="0 title"
      name="title"
      onChange={[Function]}
      type="text"
      value="new quiz" />
    <section>
      <div
        className="delete">
        <button
          className="deleteQuiz"
          id="deleteQuiz"
          onClick={[Function]}>
          Delete
        </button>
      </div>
      <div>
        <div
          className="passing-score">
          <h3>
            Passing Score
          </h3>
          <input
            className="edit-minimumScore"
            id="0 minimumScore"
            name="minimumScore"
            onChange={[Function]}
            type="number"
            value={0} />
        </div>
      </div>
      <div
        className="save">
        <button
          className="updateQuiz"
          id="updateQuiz"
          onClick={[Function]}>
          Save
        </button>
      </div>
    </section>
  </div>
  <div
    className="editQuiz-item">
    <h3>
      Question
    </h3>
    <div
      className="question-container">
      <textarea
        className="edit-question"
        id="0 question"
        name="itemQuestion"
        onChange={[Function]}
        type="text"
        value="enter question here" />
    </div>
    <h3
      className="answers">
      Answers
    </h3>
    <div
      className="answer-container"
      id="0 answerDiv 0">
      <textarea
        className="edit-quiz-item"
        id="0 answer 0"
        name="itemAnswer"
        onChange={[Function]}
        type="text"
        value="enter answer here" />
      <div
        className="buttons-container">
        <button
          className="modify-answers"
          id="0 answerCorrect 0"
          onClick={[Function]}>
          True
        </button>
        <button
          className="modify-answers"
          id="0 deleteAnswer 0"
          onClick={[Function]}>
          Delete
        </button>
      </div>
    </div>
    <div
      className="answer-container"
      id="0 answerDiv 1">
      <textarea
        className="edit-quiz-item"
        id="0 answer 1"
        name="itemAnswer"
        onChange={[Function]}
        type="text"
        value="enter answer here" />
      <div
        className="buttons-container">
        <button
          className="modify-answers"
          id="0 answerCorrect 1"
          onClick={[Function]}>
          False
        </button>
        <button
          className="modify-answers"
          id="0 deleteAnswer 1"
          onClick={[Function]}>
          Delete
        </button>
      </div>
    </div>
    <div
      className="answer-container"
      id="0 answerDiv 2">
      <textarea
        className="edit-quiz-item"
        id="0 answer 2"
        name="itemAnswer"
        onChange={[Function]}
        type="text"
        value="enter answer here" />
      <div
        className="buttons-container">
        <button
          className="modify-answers"
          id="0 answerCorrect 2"
          onClick={[Function]}>
          False
        </button>
        <button
          className="modify-answers"
          id="0 deleteAnswer 2"
          onClick={[Function]}>
          Delete
        </button>
      </div>
    </div>
    <div
      className="answer-container"
      id="0 answerDiv 3">
      <textarea
        className="edit-quiz-item"
        id="0 answer 3"
        name="itemAnswer"
        onChange={[Function]}
        type="text"
        value="enter answer here" />
      <div
        className="buttons-container">
        <button
          className="modify-answers"
          id="0 answerCorrect 3"
          onClick={[Function]}>
          False
        </button>
        <button
          className="modify-answers"
          id="0 deleteAnswer 3"
          onClick={[Function]}>
          Delete
        </button>
      </div>
    </div>
    <div
      className="item-buttons">
      <button
        className="modify-answers"
        id="0 addAnswer"
        onClick={[Function]}>
        Add Answer
      </button>
      <button
        className="modify-answers"
        id="0 deleteItem"
        onClick={[Function]}>
        Delete Question
      </button>
    </div>
  </div>
  <div
    className="addItem">
    <button
      className="addItem-button"
      id="0 addItem"
      name="0 addItem"
      onClick={[Function]}>
      Add Question
    </button>
  </div>
</div>
`;
